`config`: add `iceberg_rest_catalog_authentication_mode`

WillemKauf opened the pull request:
Add `iceberg_rest_catalog_authentication_mode` and `iceberg_rest_catalog_oauth2_server_uri` configuration options.

## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes


* none

WillemKauf committed:
`iceberg`: add `oauth2_server_uri` to `credentials`

WillemKauf committed:
`config`: add `iceberg_rest_catalog_authentication_mode`

For configuring the authentication method used for the `catalog_client`
requests made to the catalog.

There are three possible values for this `enum`:

  1. `::none`, where no authentication is attempted.
  2. `::bearer`, where the token provided via `iceberg_rest_catalog_token`
      is used for authentication.
  3. `::oauth2`, where the client credentials defined via
     `iceberg_rest_catalog_client_id/secret` are used to retrieve
      an access token for future use.

WillemKauf committed:
`config`: add `iceberg_rest_catalog_oauth2_server_uri`

For allowing the user to specify a uri to an OAuth2 server
for use in authenticating requests made by the `catalog_client`
to the catalog.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/61413-->
<!--0467fdaa-3134-468c-b06f-c9bd6252ecbe-->
<!--08155775-d657-431b-845c-4ee335c826f9-->
<!--title-->
## Retry command for [Build#61413](https://buildkite.com/redpanda/redpanda/builds/61413)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```

/ci-repeat 1
tests/rptest/tests/polaris_catalog_smoke_test.py::PolarisCatalogSmokeTest.test_connecting_to_catalog@{"cloud_storage_type":1,"with_tls":false}
tests/rptest/tests/polaris_catalog_smoke_test.py::PolarisCatalogSmokeTest.test_connecting_to_catalog@{"cloud_storage_type":1,"with_tls":true}
```
<!--/body-->

WillemKauf committed:
`iceberg`: use `oauth2_server_uri` in `catalog_client`

WillemKauf committed:
`iceberg`: add `maybe_add_bearer_auth()` in `catalog_client`

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--c31f50b0-1be6-4d52-85a4-865756a6c9b5-->
<!--TEST RESULTS-->
<!--2649c282-e327-4a7e-b27c-fe51d72c4fcf-->
<!--TEST RESULTS-->
<!--f908d5b6-07e5-414b-833e-678cc9c0f52b-->
<!--TEST RESULTS-->
<!--963550aa-cdb9-41ce-b3ac-450704ea86c0-->
<!--TEST RESULTS-->
<!--0b3b3768-43f9-4cf0-98b4-5577f10cd0ee-->
<!--TEST RESULTS-->
<!--021a88f9-d8ef-4aaa-b866-8421d206e169-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61423">build#61423</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| gtest_raft_rpunit.gtest_raft_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/61423#0194ba4c-385f-48a1-bd80-40a25c387208 | FLAKY | 1/2 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61423#0194baa8-73b0-4829-b1fe-6bd344b0b466 | FLAKY | 1/2 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61423#0194baac-12d6-4de7-be47-f79878594dbc | FLAKY | 1/3 |
| rptest.tests.datalake.compaction_test.CompactionGapsTest.test_translation_no_gaps.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61423#0194baac-12d7-4f9b-b2b5-f68739b0ee24 | FLAKY | 1/2 |
| rptest.tests.partition_force_reconfiguration_test.NodeWiseRecoveryTest.test_node_wise_recovery.dead_node_count=1 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61423#0194baa8-73b0-4829-b1fe-6bd344b0b466 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61438">build#61438</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61438#0194bce7-8113-4b07-bac1-0261019fff89 | FLAKY | 1/2 |
| rptest.tests.consumer_group_balancing_test.ConsumerGroupBalancingTest.test_coordinator_nodes_balance | ducktape | https://buildkite.com/redpanda/redpanda/builds/61438#0194bce7-8115-4648-bc9a-a9632dc84135 | FLAKY | 1/2 |
| rptest.tests.datalake.simple_connect_test.RedpandaConnectIcebergTest.test_translating_avro_serialized_records.cloud_storage_type=CloudStorageType.S3.scenario=simple | ducktape | https://buildkite.com/redpanda/redpanda/builds/61438#0194bce3-b648-4de6-81d6-0a94cf40d184 | FLAKY | 1/2 |
| rptest.tests.internal_topic_protection_test.InternalTopicProtectionLargeClusterTest.test_consumer_offset_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/61438#0194bce7-8115-4648-bc9a-a9632dc84135 | FLAKY | 1/2 |
| rptest.tests.partition_movement_test.SIPartitionMovementTest.test_shadow_indexing.num_to_upgrade=0.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/61438#0194bce7-8115-4648-bc9a-a9632dc84135 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61474">build#61474</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:--------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade | ducktape | https://buildkite.com/redpanda/redpanda/builds/61474#0194bf68-19f3-443a-aa71-3c6360d84182 | FLAKY | 1/2 |
| rptest.tests.datalake.datalake_e2e_test.DatalakeE2ETests.test_topic_lifecycle.cloud_storage_type=CloudStorageType.S3.filesystem_catalog_mode=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/61474#0194bf71-acf6-446d-aebf-c791682c9ad4 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61885">build#61885</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61885#01950775-b1d7-4ed7-9126-77a442974ed7 | FLAKY | 1/6 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61885#01950789-96c8-4bcc-ad21-698235ef7662 | FLAKY | 1/4 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61885#01950775-b1d7-4ed7-9126-77a442974ed7 | FLAKY | 1/3 |
| rptest.tests.partition_movement_test.SIPartitionMovementTest.test_shadow_indexing.num_to_upgrade=0.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/61885#01950775-b1d7-4ed7-9126-77a442974ed7 | FLAKY | 1/2 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/61885#01950775-b1d7-4ed7-9126-77a442974ed7 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61888">build#61888</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61888#01950a31-e3bb-4de6-92a1-63ed11069502 | FLAKY | 1/2 |
| rptest.tests.delete_records_test.DeleteRecordsTest.test_delete_records_with_transactions.cloud_storage_enabled=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/61888#01950a2e-ca93-4c66-b19f-8174742d626b | FLAKY | 1/2 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.delete.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=10 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61888#01950a31-e3bb-4de6-92a1-63ed11069502 | FLAKY | 1/2 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61888#01950a31-e3bb-4de6-92a1-63ed11069502 | FLAKY | 1/2 |
| rptest.tests.partition_movement_test.SIPartitionMovementTest.test_shadow_indexing.num_to_upgrade=0.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/61888#01950a31-e3bb-4de6-92a1-63ed11069502 | FLAKY | 1/3 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62090">build#62090</a></summary>


| test_id                                                                                                                                          | test_kind   | job_url                                                                                   | test_status   | passed   |
|:-------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.cloud_storage_timing_stress_test.CloudStorageTimingStressTest.test_cloud_storage_with_partition_moves.cleanup_policy=compact.delete | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62090#019526f9-0882-4256-9aa4-e43b45a59c63 | FLAKY         | 1/2      |
</details>
<!--/body-->

dotnwat commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,26 @@ configuration::configuration()
+      "refresh the token will be made. In `oauth2` mode, the credentials "
+      "(which must be) provided by `iceberg_rest_catalog_client_id` and "
+      "`iceberg_rest_catalog_client_secret` are used to obtain a bearer token "
+      "from the uri specified by `iceberg_rest_catalog_oauth2_server_uri`. If "
```
maybe this commit should come after the commit which adds `iceberg_rest_catalog_oauth2_server_uri`?

WillemKauf replied:
Moved.

dotnwat commented on src/v/datalake/coordinator/catalog_factory.cc:
```diff
@@ -137,12 +138,35 @@ filesystem_catalog_factory::create_catalog() {
   : config_(&config) {}
 
+void rest_catalog_factory::maybe_make_credentials_or_token(
+  std::optional<iceberg::rest_client::credentials>& credentials,
+  std::optional<iceberg::rest_client::oauth_token>& token) {
```
should we just return a struct or pair?

WillemKauf replied:
Sure, happy to change this from in/out parameters to a return type if that would be more ergonomic.

dotnwat replied:
we generally don't use out-parameters unless the isn't a good alternative. `struct credentials_or_token { ... };`

dotnwat commented on src/v/iceberg/tests/rest_catalog_test.cc:
```diff
@@ -67,7 +68,10 @@ struct RestCatalogTest
-          iceberg::rest_client::api_version("v1"));
+          iceberg::rest_client::api_version("v1"),
+          std::nullopt,
+          nullptr,
+          config::datalake_catalog_auth_mode::oauth2);
```
doesn't this commit need to be merged into the commit which changes the catalog_client interface?

WillemKauf replied:
Yes, it can be squashed with commit before it.

WillemKauf commented:
Force push to:
* Move commit history around for config properties
* Squash commit for `rest_catalog_test`

WillemKauf committed:
`datalake`: add `make_credentials_or_token()`

Depending on authentication settings, we may either construct
user credentials for use with the OAuth2 endpoint for retrieving a
token, construct the token directly, or do nothing at all.

WillemKauf commented:
Force push to:
* change `maybe_make_credentials_or_token()` to `make_credentials_and_token()`

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+      "iceberg_rest_catalog_oauth2_server_uri",
+      "The OAuth URI used for retrieving access tokens for catalog "
+      "authentication. The path `{oauth_server_uri}/oauth/tokens` will be used "
+      "for retrieving tokens. If unset, the deprecated catalog path will be "
```
```suggestion
      "for retrieving tokens. If unset, the deprecated catalog path is used "
```

asimms41 replied:
```suggestion
      "If unset, the deprecated catalog path is "
```

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+  , iceberg_rest_catalog_oauth2_server_uri(
+      *this,
+      "iceberg_rest_catalog_oauth2_server_uri",
+      "The OAuth URI used for retrieving access tokens for catalog "
```
```suggestion
      "The OAuth URI used for retrieving access tokens for the Iceberg catalog "
```

asimms41 replied:
```suggestion
      "OAuth URI used to retrieve access tokens for Iceberg catalog "
```

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+  , iceberg_rest_catalog_authentication_mode(
+      *this,
+      "iceberg_rest_catalog_authentication_mode",
+      "The authentication mode used for client requests made to the Iceberg "
```
```suggestion
      "The authentication mode for client requests made to the Iceberg "
```

asimms41 replied:
```suggestion
      "The authentication mode for client requests made to the Iceberg "
```

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+      "iceberg_rest_catalog_authentication_mode",
+      "The authentication mode used for client requests made to the Iceberg "
+      "catalog. You can choose between: `none`, `bearer`, and `oauth2`. In "
+      "`bearer` mode, the token (which must be) provided by "
```
```suggestion
      "`bearer` mode, the token specified in "
```

asimms41 replied:
```suggestion
      "`bearer` mode, the token specified in "
```

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+      *this,
+      "iceberg_rest_catalog_oauth2_server_uri",
+      "The OAuth URI used for retrieving access tokens for catalog "
+      "authentication. The path `{oauth_server_uri}/oauth/tokens` will be used "
```
```suggestion
      "authentication. When this property is set, tokens are retrieved from the path`{oauth_server_uri}/oauth/tokens`."
```

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+      *this,
+      "iceberg_rest_catalog_authentication_mode",
+      "The authentication mode used for client requests made to the Iceberg "
+      "catalog. You can choose between: `none`, `bearer`, and `oauth2`. In "
```
```suggestion
      "catalog. Choose from: `none`, `bearer`, and `oauth2`. In "
```

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+      "The authentication mode used for client requests made to the Iceberg "
+      "catalog. You can choose between: `none`, `bearer`, and `oauth2`. In "
+      "`bearer` mode, the token (which must be) provided by "
+      "`iceberg_rest_catalog_token` is used unconditonally. No attempts to "
```
```suggestion
      "`iceberg_rest_catalog_token` is used unconditionally, and no attempts are made to "
```

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+      "catalog. You can choose between: `none`, `bearer`, and `oauth2`. In "
+      "`bearer` mode, the token (which must be) provided by "
+      "`iceberg_rest_catalog_token` is used unconditonally. No attempts to "
+      "refresh the token will be made. In `oauth2` mode, the credentials "
```
```suggestion
      "to refresh the token. In `oauth2` mode, the credentials "
```

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+      "`bearer` mode, the token (which must be) provided by "
+      "`iceberg_rest_catalog_token` is used unconditonally. No attempts to "
+      "refresh the token will be made. In `oauth2` mode, the credentials "
+      "(which must be) provided by `iceberg_rest_catalog_client_id` and "
```
```suggestion
      "specified in `iceberg_rest_catalog_client_id` and "
```

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+      "refresh the token will be made. In `oauth2` mode, the credentials "
+      "(which must be) provided by `iceberg_rest_catalog_client_id` and "
+      "`iceberg_rest_catalog_client_secret` are used to obtain a bearer token "
+      "from the uri specified by `iceberg_rest_catalog_oauth2_server_uri`. If "
```
```suggestion
      "from the URI specified by `iceberg_rest_catalog_oauth2_server_uri`. If "
```

asimms41 commented on src/v/config/configuration.cc:
```diff
@@ -3792,6 +3792,36 @@ configuration::configuration()
+      "(which must be) provided by `iceberg_rest_catalog_client_id` and "
+      "`iceberg_rest_catalog_client_secret` are used to obtain a bearer token "
+      "from the uri specified by `iceberg_rest_catalog_oauth2_server_uri`. If "
+      "this server uri is left undefined, the deprecated catalog endpoint "
```
```suggestion
      "this server URI is left undefined, the deprecated Iceberg catalog endpoint "
```

WillemKauf commented:
Force push to:
* Rebase to `upstream/dev` to fix merge conflicts

WillemKauf commented:
Force push to:
* Make changes to `config/configuration.cc` as requested by docs team.

dotnwat commented:
is this ready for review, i can't tell

WillemKauf commented:
>is this ready for review, i can't tell

Yes

andrwng commented on src/v/config/configuration.cc:
```diff
@@ -3822,6 +3822,26 @@ configuration::configuration()
+      "`iceberg_rest_catalog_client_secret` are used to obtain a bearer token "
+      "from the URI specified by `iceberg_rest_catalog_oauth2_server_uri`. If "
+      "this server URI is left undefined, the deprecated Iceberg catalog "
+      "endpoint `/v1/oauth/tokens` is used instead.",
```
>If this server URI is left undefined, the deprecated Iceberg catalog endpoint `/v1/oauth/tokens` is used instead.

nit: maybe remove this bit to avoid duplicating, given it's already covered in the description of that property

andrwng commented on tests/rptest/tests/cluster_config_test.py:
```diff
@@ -1705,6 +1705,38 @@ def test_disable_bounded_property_checks(self):
+            'panda_secret'
+        }]
+
+        for valid_props in valid_auth_mode_props:
```
Consider adding some cases for invalid configs

WillemKauf replied:
Added

andrwng commented on src/v/iceberg/rest_client/catalog_client.cc:
```diff
@@ -104,15 +104,17 @@ catalog_client::catalog_client(
 
 ss::future<expected<oauth_token>>
 catalog_client::acquire_token(retry_chain_node& rtc) {
+    const auto& creds = _credentials.value();
```
What guarantees that this isn't nullopt here?

WillemKauf replied:
(Squashed commit which adds a `vassert()` here into this commit, hope that makes the unchecked `.value()` more digestible)

But, in general, this code path is only reached from `ensure_token()`, which is only reached when the `_auth_mode` is set to `oauth2`, which is also the condition used to determine if `credentials` should have a value when `catalog_client` is being constructed.

So, we are relying on cluster property validation to ensure this value is always present here.

andrwng commented on src/v/datalake/coordinator/catalog_factory.cc:
```diff
@@ -206,7 +206,7 @@ rest_catalog_factory::create_catalog() {
       std::move(prefix_path),                             // prefix
       std::nullopt,                                       // api_version
-      std::move(creds_and_token.token)                    // token
+      std::move(creds_and_token.token),                   // token
```
nit: for the previous commit?

WillemKauf replied:
Squashed

andrwng commented on src/v/iceberg/rest_client/catalog_client.cc:
```diff
@@ -112,10 +113,18 @@ catalog_client::acquire_token(retry_chain_node& rtc) {
+    // Otherwise, fall back to the deprecated /oauth/tokens catalog endpoint.
+    ss::sstring token_path
+      = creds.oauth2_server_uri.has_value()
+          ? ss::sstring{fmt::format(
+              "{}/{}", creds.oauth2_server_uri.value(), oauth_token_endpoint)}
```
Curious, why not make the user specify the full path? Wondering if the oauth/tokens suffix is ever undesirable

WillemKauf replied:
I have no real preference either way. WDYT? 

Maybe we could allow the user to either specify it in the URI or not, and then guarantee it is there when using the URI by adding it, or leaving the URI untouched.

andrwng replied:
Without much context around auth in general, it feels more future proof to require users to provide the full endpoint. Hard-coding /oauth/tokens before seemed fine because at least it was part of the REST spec, but now it doesn't seem well-defined that providers will always use that suffix

andrwng commented on src/v/iceberg/rest_client/catalog_client.cc:
```diff
@@ -158,6 +159,35 @@ catalog_client::ensure_token(retry_chain_node& rtc) {
     co_return _oauth_token->access_token;
 }
 
+ss::future<std::optional<domain_error>> catalog_client::maybe_add_bearer_auth(
```
Can we use expected<nullopt_t> here or somesuch? It's really error prone to mix in an optional as the error type, given has_value() means the opposite it does elsewhere.

WillemKauf replied:
Not sure why I did this to begin with!

Flipped to `expected<std::monostate>`.

andrwng commented on tests/rptest/tests/polaris_catalog_smoke_test.py:
```diff
@@ -77,6 +77,8 @@ def _start_redpanda(self, catalog_prefix, client_id, client_secret,
             "rest",
             "iceberg_rest_catalog_endpoint":
             self.polaris.catalog_url,
+            "iceberg_rest_catalog_authentication_mode":
```
Please add commit messages.

Is this added just to exercise the new code? Or was this necessary? If so, can this be squashed into the functional commit that made it necessary

andrwng reviewed (commented):
Nice

WillemKauf commented:
Force push to: 
* Change `std::optional<domain_error>` to `expected<std::monostate>` in `catalog_client::maybe_add_bearer_auth()`
* Add invalid test cases to `cluster_config_test.py`
* Clean up commit history by squashing some commits together

WillemKauf commented:
I'm going to leave parameterized Nessie testing as work for a follow-up PR.

WillemKauf commented:
Force push to:
* Remove `oauth/tokens` suffix appending to `oauth2_server_uri`. Users must specified the full token path to `iceberg_rest_catalog_oauth2_server_uri`.

andrwng commented on src/v/config/configuration.cc:
```diff
@@ -3812,6 +3812,36 @@ configuration::configuration()
+      "iceberg_rest_catalog_oauth2_server_uri",
+      "The OAuth URI used to retrieve access tokens for Iceberg catalog "
+      "authentication. When this property is set, tokens are retrieved from "
+      "the path `{oauth_server_uri}/oauth/tokens`. If unset, the deprecated "
```
nit: needs an update

WillemKauf replied:
Oops. Sorry!

WillemKauf commented:
Force push to:
* Update configuration descriptions
